src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -80,7 +80,7 @@ constructor a_fresh statements
 
   goS env s
    = let env' = updateExpEnv s env
-         go   = goX env
+         go   = goX env'
```

Comment:
yeah, this shouldn't have the prime. if you have something like
```
Let x = 5
Let x = (x + 2)
```
here we need to be very careful that the x mentioned in "x + 2" is the "x=5", not the "x=x+2".
(I'm not certain how adding the prime would make it more effective anyway)